lib/bugsnag/middleware/ignore_error_class.rb

```diff
@@ -10,9 +10,16 @@ def initialize(bugsnag)
     def call(report)
       ignore_error_class = report.raw_exceptions.any? do |ex|
         ancestor_chain = ex.class.ancestors.select { |ancestor| ancestor.is_a?(Class) }.to_set
+        ancestor_chain_strings = ancestor_chain.map(&:to_s)
```

Comment:
Having seen this in the test, I wonder whether matching strings and regexes against ancestors is confusing. I can see if you pass an actual class in then matching subclasses sounds reasonable, but with strings and particularly regexes I think we should probably just match against the exception class. But I guess it could be argued that it would be confusing to have two different behaviours...

Reply:
Yeah I can see the arguments both ways

I went for this way because the migration path to fix the Rails deprecation is as easy as wrapping the current values in quotes

If we do an exact match then you'd need to find out each concrete exception class you want to ignore if you were relying on ignoring an ancestor using a class constant

Then because of that, `Regexp` being the only value handled differently would feel weird, though I can see how it might be very unexpected to match against every ancestor